File: src/main/java/ch/uzh/ifi/access/model/dto/TaskFileDTO.java
package ch.uzh.ifi.access.model.dto;

import ch.uzh.ifi.access.model.constants.Context;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class TaskFileDTO {
    Long id;
    Long templateId;
    String templatePath;
    String path;
    Context context;
    boolean editable;
    boolean enabled = true;
}

File: src/main/kotlin/ch/uzh/ifi/access/model/dto/TaskFileDTO.kt
package ch.uzh.ifi.access.model.dto

import lombok.Data
import lombok.NoArgsConstructor

@Data
@NoArgsConstructor
class TaskFileDTO {
    var path: String? = null
    var template: String? = null
    // refers to the programming language, e.g. "py"
    var language: String? = null
    var binary = false
}
